openshift/templates/postgresql/postgresql-deploy.yaml

```diff
@@ -6,6 +6,65 @@ metadata:
     tags: ${NAME}
   name: ${NAME}-persistent-template
 objects:
+  # - kind: NetworkPolicy
+  #   apiVersion: networking.k8s.io/v1
+  #   metadata:
+  #     name: ${NAME}-allow-service
+  #     labels:
+  #       name: ${NAME}-allow-agent
+  #       app: ${APP_NAME}
+  #       env: ${TAG_NAME}
+  #   spec:
+  #     description: Allow the service to access the database.
+  #     ingress:
+  #       - from:
+  #         # - podSelector:
+  #         #     matchLabels:
+  #         #       role: service
+  #         #       app: ${APP_NAME}
+  #         #       env: ${TAG_NAME}
+  #         - namespaceSelector:
+  #             matchLabels:
+  #               name: ${NAMESPACE_NAME}
+  #               environment: ${TAG_NAME}
+  #         # ports:
+  #         #   - protocol: TCP
+  #         #     port: 5432
+  #     podSelector:
+  #       matchLabels:
+  #         role: ${ROLE}
+  #         app: ${APP_NAME}
+  #         env: ${TAG_NAME}
+  - kind: NetworkPolicy
+    apiVersion: networking.k8s.io/v1
+    metadata:
+      name: ${NAME}
+      labels:
+        name: ${NAME}
+        app: ${APP_NAME}
+        env: ${TAG_NAME}
+    spec:
+      description: |
+        Allow the api(s) to access the db.
+      ingress:
+        - from:
+            - podSelector:
+                matchLabels:
+                  role: service
```

Comment:
The role of the service can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.